### ISSUE TITLE
Unable to create new user with homed

### ISSUE THREAD

**Issue by jjcasmar** [systemd/systemd#29160]
### systemd version the issue has been seen with

254

### Used distribution

Archlinux

### Linux kernel version used

6.5.2-arch1-1

### CPU architectures issue was seen on

x86_64

### Component

systemd-homed

### Expected behaviour you didn't see

I am trying to create a new user, but fails to be created

### Unexpected behaviour you saw

```
sudo homectl create prueba
Please enter new password for user prueba: ****                    
Please enter new password for user prueba (repeat): ****                    
Operation on home prueba failed: Failed to execute operation: Protocol error
```

journalctl shows the following

```
Sep 11 16:29:30 archlinux systemd-homed[612]: prueba: changing state absent → creating
Sep 11 16:29:30 archlinux systemd-homework[13236]: Using automatic default storage of 'luks'.
Sep 11 16:29:30 archlinux systemd-homework[13236]: Sizing home to 83% of available disk space, which is 346.0G.
Sep 11 16:29:30 archlinux systemd-homework[13236]: Allocating image file completed.
Sep 11 16:29:30 archlinux systemd-homework[13236]: Writing of partition table completed.
Sep 11 16:29:30 archlinux systemd-homework[13236]: Setting up loopback device /dev/loop1 completed.
Sep 11 16:29:31 archlinux systemd-homework[13236]: LUKS formatting completed.
Sep 11 16:29:33 archlinux systemd-homework[13236]: Writing password to LUKS keyslot 0 completed.
Sep 11 16:29:33 archlinux systemd-homework[13236]: LUKS activation by volume key succeeded.
Sep 11 16:29:33 archlinux systemd-homework[13236]: Writing user record as LUKS token completed.
Sep 11 16:29:33 archlinux systemd-homework[13236]: Setting up LUKS device /dev/mapper/home-prueba completed.
Sep 11 16:29:33 archlinux systemd-homed[13246]: mkfs.btrfs: error while loading shared libraries: cannot make segment writable for relocation: Permission denied
Sep 11 16:29:33 archlinux systemd-homework[13236]: (mkfs) failed with exit status 127.
Sep 11 16:29:33 archlinux systemd-homed[612]: block device /sys/devices/virtual/block/dm-1 has been removed.
Sep 11 16:29:33 archlinux systemd-homed[612]: block device /sys/devices/virtual/block/dm-1 has been removed.
Sep 11 16:29:33 archlinux systemd-homed[612]: block device /sys/devices/virtual/block/loop1 has been removed.
Sep 11 16:29:33 archlinux systemd-homed[612]: Operation on prueba failed: Protocol error
```

I can create new users using `--storage=subvolume`

**Comment by poettering** [comment]
Hmm, Arch seems to build those btrfs binaries in a way they are incompatible with MemoryDenyWriteExecutable= (i.e. PR_SET_MDWE). 

This can be worked around locally, by dropping that setting from the unit file.

But this ultimately needs to be fixed in the archlinux toolchain, they really should make their stuff work with PR_SET_MDWE enabled.

**Comment by mrc0mmand** [comment]
I opened https://bugs.archlinux.org/task/79654 for btrfs-progs, since it seems to happen only with mkfs.btrfs (at least from what I've tested so far).

**Comment by mrc0mmand** [comment]
Should be fixed in btrfs-progs 6.5.1.
